### PR TITLE
add a new string representation: '{string, list()}' tuple

### PR DESCRIPTION
**Pull request by petrkozorezov**

### CODE REVIEW

File: src/msgpack.erl [hunk 1]
```diff
@@ -48,13 +48,13 @@
 -export_type([object/0, msgpack_map/0, options/0, ext_packer/0, ext_unpacker/0]).
 -type object() :: msgpack_term().
 
--type options() :: 
+-type options() ::
         [{spec, new|old} |
          {allow_atom, none|pack} |
          {known_atoms, [atom()]} |
-         {unpack_str, as_binary|as_list} |
+         {unpack_str, as_binary|as_list|with_tag} |
```
Comment (keynslug) [inline src/msgpack.erl]: `as_tagged_list`?

File: src/msgpack_unpacker.erl [hunk 2]
```diff
@@ -233,7 +233,8 @@ unpack_str_or_raw(V, ?OPTION{spec=new,
     {case UnpackStr of
          as_binary when ValidateString -> unpack_str(V), maybe_bin(V, Opt);
          as_binary -> maybe_bin(V, Opt);
-         as_list -> unpack_str(V)
+         as_list -> unpack_str(V);
+         with_tag -> {string, unpack_str(V)}
```
Comment (keynslug) [inline src/msgpack_unpacker.erl]: Why would we want tagged list/string representation instead of, say, tagged binary?

File: src/msgpack.erl [hunk 3]
```diff
@@ -48,13 +48,13 @@
 -export_type([object/0, msgpack_map/0, options/0, ext_packer/0, ext_unpacker/0]).
```
Comment (keynslug) [inline src/msgpack.erl:48]: Should we bump the app version or is it maintainer's responsibility?

File: src/msgpack.hrl [hunk 4]
```diff
@@ -37,9 +37,9 @@
           spec = new :: new | old,
```
Comment (keynslug) [inline src/msgpack.hrl:37]: 1. Shouldn't we update `msgpack_term()` spec to include tagged lists/strings?
2. Someone forgot to add `positive_infinity` and the like to the spec earlier.

File: src/msgpack.erl (hunk 3 above)
Comment (petrkozorezov) [reply src/msgpack.erl:48]: I think, a maintainer will do it. 😉

File: src/msgpack_unpacker.erl (hunk 2 above)
Comment (petrkozorezov) [reply src/msgpack_unpacker.erl]: Because if we use the string type, we want a codepoints list.

File: src/msgpack.hrl (hunk 4 above)
Comment (petrkozorezov) [reply src/msgpack.hrl:37]: 1. 👍 
2. it should be an another PR

**Review by keynslug — APPROVED** [review]

**Pull request merged**